CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+# Contributing
+
+Contributions to GT4Py are welcome and greatly appreciated. Proper credit will be given to contributors by adding their names to the [AUTHORS.md](AUTHORS.md) file. Note that [ETH Zurich](https://ethz.ch/en.html) is the owner of the GridTools project and the GT4Py library, therefore external contributors must sign a contributor assignment agreement.
+
+## Types of Contributions
+
+### Report Bugs
+
+Report bugs at [https://github.com/gridtools/gt4py/issues](https://github.com/gridtools/gt4py/issues).
+
+If you are reporting a bug, please include:
+
+- Your operating system name and version.
+- Python interpreter version.
+- Any details about your local setup that might be helpful in troubleshooting.
+- Detailed steps to reproduce the bug.
+
+### Fix Bugs
+
+Look through the [GitHub issues](https://github.com/gridtools/gt4py/issues) for bugs. Anything tagged with `bug` and `help wanted` is open to whomever wants to implement it.
+
+### Implement Features
+
+Look through the [GitHub issues](https://github.com/gridtools/gt4py/issues) for features. Anything tagged with `enhancement` and `help wanted` is open to whomever wants to implement it.
+
+### Write Documentation
+
+GT4Py could always use more documentation, whether as part of the official GT4Py docs, in docstrings, or even on the web in blog posts and articles.
+
+### Submit Feedback
+
+The best way to send feedback is to file an issue at [https://github.com/gridtools/gt4py/issues](https://github.com/gridtools/gt4py/issues).
+
+If you are proposing a feature:
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Remember that this is an open-source project and that contributions
+  are welcome :)
+
+## Getting Started
+
+Ready to start contributing? We use a [fork and pull request](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow) workflow for contributions. Pull requests need to pass all the automated checks as well as a review before they can be merged. To set up properly your local development environment follow these steps:
+
+1. Fork the [GT4Py](https://github.com/gridtools/gt4py) repo on GitHub.
+
+2. Clone your fork locally and check out the relevant branch:
+
+   ```bash
+   $ git clone git@github.com:your_name_here/gt4py.git
+   $ cd gt4py
+   $ git checkout functional
```

Comment:
```suggestion
   $ git checkout main
```

Or if we want it generic: `<branch name>`. Alternatively, we could drop the checkout, since everything should be going into main now...